cmd/dp-deployer/main.go

```diff
@@ -53,17 +78,55 @@ func main() {
 		os.Exit(1)
 	}
 
+	//TODO: remove this when config is not driven by flags
+	healthInterval, err := time.ParseDuration(*healthcheckInterval)
+	if err != nil {
+		log.Error(err, nil)
+	}
+
+	healthTimeout, err := time.ParseDuration(*healthcheckCriticalTimeout)
+	if err != nil {
+		log.Error(err, nil)
+	}
+
+	hcc := &healthcheckConfig{
+		HealthcheckInterval:        healthInterval,
+		HealthcheckCriticalTimeout: healthTimeout,
+		BindAdd:                    *healthcheckPort,
+	}
+
+	// Create healthcheck object with versionInfo
+	versionInfo, err := healthcheck.NewVersionInfo(BuildTime, GitCommit, Version)
+	if err != nil {
+		log.Error(err, nil)
+		os.Exit(1)
+	}
+	hc := healthcheck.New(versionInfo, hcc.HealthcheckCriticalTimeout, hcc.HealthcheckInterval)
+
+	r := mux.NewRouter()
+	r.HandleFunc("/health", hc.Handler)
+
+	// Start healthcheck
+	hc.Start(ctx)
+
 	sigC := make(chan os.Signal)
 	signal.Notify(sigC, syscall.SIGHUP, syscall.SIGINT, syscall.SIGQUIT, syscall.SIGTERM)
 
-	ctx, cancel := context.WithCancel(context.Background())
-
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
 		e.Start(ctx)
 	}()
 
+	// Create and start http server for healthcheck
+	httpServer := server.New(hcc.BindAdd, r)
+	go func() {
+		if err := httpServer.ListenAndServe(); err != nil {
+			log.ErrorCtx(ctx, err, log.Data{"config": hcc})
+			os.Exit(2)
```

Comment:
Sounds like I'm being contrary, but this exit makes me uncomfortable, because it's inside a goroutine
```suggestion
			cancel()
```